src/main/java/de/tum/in/www1/exerciseapp/service/CourseService.java

```diff
@@ -105,4 +103,68 @@ public void delete(Long id) {
         List<Course> courses = courseRepository.findAll();
         return courses.stream().map(c -> c.getTeachingAssistantGroupName()).collect(Collectors.toList());
     }
+
+    /**
+     * Getting a Collection of Results in which the average Score of all taken Exercises, is mapped to a userId for the instructor dashboard
+     *
+     * @param id the courseId
+     *
+     * @return The resultId refers in this case to the UserID to whom the overallScore belongsTo
+     */
+    @Transactional(readOnly = true)
+    public Collection<Result> getAllSummedOverallScoresOfCourse(Long id){
+        Course course = findOne(id);
+        Set<Exercise> exercisesOfCourse = course.getExercises();
+        HashMap<Long, Result> allOverallScoresOfCourse = new HashMap<>();
```

Comment:
the code looks much cleaner and is easier to understand, but more comments and explanations would help ;-) please add more details, e.g. what does the Long in 'allOverallScoresOfCourse' represents?

Reply:
changed id to courseId should help to improve further understanding